velox/exec/benchmarks/HashJoinListResultBenchmark.cpp

```diff
@@ -0,0 +1,476 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/common/base/SelectivityInfo.h"
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/common/memory/MmapAllocator.h"
+#include "velox/exec/HashTable.h"
+#include "velox/exec/OperatorUtils.h"
+#include "velox/exec/VectorHasher.h"
+#include "velox/functions/prestosql/aggregates/RegisterAggregateFunctions.h"
+#include "velox/vector/tests/utils/VectorTestBase.h"
+
+#include <folly/Benchmark.h>
+#include <folly/executors/CPUThreadPoolExecutor.h>
+#include <folly/init/Init.h>
+#include <gtest/gtest.h>
+#include <memory>
+
+using namespace facebook::velox;
+using namespace facebook::velox::exec;
+using namespace facebook::velox::test;
+
+namespace {
+struct HashTableBenchmarkParams {
```

Comment:
Let's extract the benchmark into a separate PR. There is a lot of code here. It will be easier to review separately.